http/http.go

```
@@ -83,12 +83,14 @@ func Call(method string, rootURL string, path string, body interface{}, response
 			traceLog.WithField("error", genErr.GetDetailString()).Error("Failed to read response body")
 			return nil, errors.NewGenericError(421, errorDomain, errorSubDomain, ErrorReadResponseBodyFailed, nil)
 		}
-
-		// Unmarshal response
-		err = json.Unmarshal(resBody, response)
-		if err != nil {
-			traceLog.WithField("error", err).Error("Failed to parse response body from JSON")
-			return nil, errors.NewGenericError(421, errorDomain, errorSubDomain, ErrorParseResponseBodyFailed, nil)
+		// just ignore if there is no answer
+		if len(resBody) > 0 {
```

When do we actually need this exception? Because I feel we should be a bit cautious with this change. In case you don't want to parse the body, you should use `http.CallRaw` instead. Also, not sure what's the impact on a `204` response code. Probably, no error is returned, but the response will also stay empty (which I assume is not something you want to go unnoticed).

---

It comes from qpark/repository/qparkAPIRepository/common.go : 

`
	if len(body) == 0 {
		return []byte{}, nil
	}
`

Looks like a good idea to me and with the http from go-utils, it broke the test. I can revert this change and change the test, but I have no idea of the impact on the FE

---

Personally, I'm a bit in doubt for this one. I checked the current code and see that the risk is quite limited, since we already check for a non-2xx return code before trying to parse the body. Therefore, the only real thing which I think would hit this is a 204 No Content.

I see 2 possible solutions:
- We change the if to `if response == nil`. This way the calling function tells the helper, I will never care about the result (but still want to use marshalling for the request).
- We remove the `if` and the calling function has to check and ignore errors with SubDomainCode `ErrorParseResponseBodyFailed`.

This way we're sure the if the calling function assumes the response is set, it's really set. This prevents weird silent errors.

---

I removed this code for now. We can discute about that laster